enhancements/authentication/automated-service-ca-rotation.md

```diff
@@ -102,29 +102,78 @@ and the functionality could be separately extracted for operator reuse.
+        the expiry of the pre-rotation CA.
+      - *R* must be greater than or equal to the minimum remaining CA
+        duration, *M*, to ensure that an upgrade occurs before a subsequent
+        rotation.
+        - Since cross-signing is only supported between the current and
+          previous CAs, rotating before all pods have been restarted to use key
+          material from the current CA risks breaking trust with key material
+          issued by the previous CA.
+    - The interval between creation and rotation, *R*, can be computed as the
+      total duration *D* less the minimum remaining duration *M*:
+      - *R* = *D* - *M*
+    - Reordering to solve for *D*:
+      - *D* = *R* + *M*
+    - Since each of *R* and *M* must be greater than *I*:
+      - *D* = *R* + *M* > 2 * *I*
+    - Since *R* >= *M* should be true, simplify to *R* = *M*:
+      - *D* = 2 * *R* > 2 * *I*
+    - Substitute *I* = 12:
+      - *D* = 2 * *R* > 24
+    - Picking *R* = *M* = 13 will satisfy the relation, resulting in *D* = 26
+  - Worst-case timelines with old and new values for CA duration:
+    - Let minimum duration *M* = 6 months, total duration *D* = 12 months:
       - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
       - T+6m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
       - T+12m - Cluster is upgraded and all pods are restarted
       - T+12m - CA-1 expires. If cluster was not upgraded before this
                 happens, services using the old key material may
                 break.
-    - With 14 month CA duration:
+    - Let minimum duration *M* = 13 months, total duration *D* = 26 months:
       - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
-      - T+8m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
       - T+12m - Cluster is upgraded and all pods are restarted
-      - T+14m - CA-1 expires. No impact because of the restart at time of upgrade
-  - Services that do not refresh key material automatically must be restarted
-    after ca rotation.
-  - Upgrades restart all pods, and ensuring that an upgrade takes place after
-    rotation and prior to expiry of the pre-rotation ca guarantees that
-    services will always be using valid key material.
-  - If the ca duration does not exceed the expected upgrade interval, there is
-    the possibility, however slight, that an upgrade will not occur after
-    automated rotation and before expiry of the pre-rotation ca.
+      - T+13m - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 13m
+      - T+24m - Cluster is upgraded and all pods are restarted
+      - T+26m - CA-1 expires. No impact because of the restart at time of upgrade
+  - If the service CA validity duration (entire, not remaining) is less than
+    26 months, then automated rotation should be triggered and the cluster
+    administrator will need to manually restart all pods.
```

Comment:
s/should/will? Wouldn't it be easier to backport the rotation to a `n-1` version so that no manual intervention is necessary and we're sure all pods contain the current certs?

Reply:
There is no such thing as 'no manual intervention' - one-time manual restart is required for any cluster upgrading to a release that supports automated rotation. There is no other way to ensure that the pre-rotation CA won't expire before a subsequent upgrade because it's remaining duration is by definition going to be less than the minimum upgrade interval of 12 months.

Reply:
I've added a new fixup intended to address your concerns, PTAL